pyproject.toml

```diff
@@ -55,6 +58,14 @@ litellm = 'litellm:run_server'
 flake8 = "^6.1.0"
 black = "^23.12.0"
 pytest = "^7.4.3"
+mypy = "^1.8.0"
```

Comment:
hey @ErikBjare @jamesbraza is it necessary to add mypy and types in our core dependencies? 

for context i'm looking at openai's dependencies and i don't see this there - https://github.com/openai/openai-python/blob/54a5911f5215148a0bdeb10e2bcfb84f635a75b9/pyproject.toml#L10

Reply:
So these are not the core deps, they're `dev` (developer) deps. Also, https://github.com/BerriAI/litellm/pull/4136 added `mypy` to the `dev` group already, so if this PR were rebased atop main this particular line would go away.

However, the below type stubs added need to stay in dev group dependencies.

Tho I don't think the type stubs should be so specifically pinned, it's okay with something like `>=` instead of `^`